src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -25,197 +26,274 @@ impl HtmlHandlebars {
+
+        // This could cause a problem when someone displays
+        // code containing <base href=...>
+        // on the front page, however this case should be very very rare...
+        content = content
+            .lines()
+            .filter(|line| !line.contains("<base href="))
+            .collect::<Vec<&str>>()
+            .join("\n");
+
+        book.write_file("index.html", content.as_bytes())?;
+
+        info!(
+            "[*] Creating index.html from {:?} ✓",
+            book.get_destination()
+                .expect(
+                    "If the HTML renderer is called, one would assume the HtmlConfig is \
+                                            set... (4)",
+                )
+                .join(&ch.path.with_extension("html"))
+        );
+
+        Ok(())
+    }
+
+    fn post_process(&self, rendered: String) -> String {
+        let rendered = build_header_links(rendered, "print.html");
+        let rendered = fix_anchor_links(rendered, "print.html");
+        let rendered = fix_code_blocks(rendered);
+        let rendered = add_playpen_pre(rendered);
+
+        rendered
+    }
+
+    fn copy_static_files(&self, book: &MDBook, theme: &Theme) -> Result<(), Box<Error>> {
+        book.write_file("book.js", &theme.js)?;
+        book.write_file("book.css", &theme.css)?;
+        book.write_file("favicon.png", &theme.favicon)?;
+        book.write_file("jquery.js", &theme.jquery)?;
+        book.write_file("highlight.css", &theme.highlight_css)?;
+        book.write_file(
+            "tomorrow-night.css",
+            &theme.tomorrow_night_css,
+        )?;
+        book.write_file(
+            "ayu-highlight.css",
+            &theme.ayu_highlight_css,
+        )?;
+        book.write_file("highlight.js", &theme.highlight_js)?;
+        book.write_file("clipboard.min.js", &theme.clipboard_js)?;
+        book.write_file("store.js", &theme.store_js)?;
+        book.write_file(
+            "_FontAwesome/css/font-awesome.css",
+            theme::FONT_AWESOME,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/fontawesome-webfont.eot",
+            theme::FONT_AWESOME_EOT,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/fontawesome-webfont.svg",
+            theme::FONT_AWESOME_SVG,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/fontawesome-webfont.ttf",
+            theme::FONT_AWESOME_TTF,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/fontawesome-webfont.woff",
+            theme::FONT_AWESOME_WOFF,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/fontawesome-webfont.woff2",
+            theme::FONT_AWESOME_WOFF2,
+        )?;
+        book.write_file(
+            "_FontAwesome/fonts/FontAwesome.ttf",
+            theme::FONT_AWESOME_TTF,
+        )?;
+
+        Ok(())
+    }
+
+    fn write_custom_file(&self, custom_file: &Path, book: &MDBook) -> Result<(), Box<Error>> {
```

Comment:
I'm pretty sure that's just a helper function which will take some path relative to the book's `src` directory and copy it to the output directory. I've added a doc-comment explaining what it does.